tasks/gov.js

```diff
@@ -47,21 +47,230 @@ task("gov:setPricingConfig")
+    .addPositionalParam("filename", "file name without .json suffix")
+    .setAction(async ({ filename }) => {
+        const fs = require("fs");
+
+        const {et, ctx, } = await setupGovernanceFork();
+
+        const transactions = await ctx.contracts.euler.queryFilter(
+            "EnterMarket",
+            "earliest",
+            "latest",
+        );
+
+        let result = [];
+        for (let i = 0; i < transactions.length; i++) {
+            let temp = {};
+            temp.account = transactions[i].args.account;
+            temp.underlying = transactions[i].args.underlying;
+            result.push(temp);
+        }
+
+        // unique event arguments based on accounts
+        const key = 'account';
+        const arrayUniqueByKey = [...new Map(result.map(item =>
+            [item[key], item])).values()];
+
+        // unique addresses regardless of markets 
+        const uniqueAddresses = [...new Set(arrayUniqueByKey.map(item => item.account))];
+
+        // compute health scores
+        let health_scores = {};
+        for (let account of uniqueAddresses) {
+            let detLiq = await ctx.contracts.exec.callStatic.detailedLiquidity(account);
+            let totalLiabilities = ethers.BigNumber.from(0);
+            let totalAssets = ethers.BigNumber.from(0);
+
+            for (let asset of detLiq) {
+                totalLiabilities = totalLiabilities.add(asset.status.liabilityValue);
+                totalAssets = totalAssets.add(asset.status.collateralValue);
+            }
+
+            let health = 0;
+            let violation = false;
+
+            if (totalAssets > 0) {
+                health = totalAssets.mul(et.c1e18);
+
+                if (totalLiabilities > 0) {
+                    health = totalAssets.mul(et.c1e18).div(totalLiabilities);
+                }
+
+                if (health.gte(et.c1e18)) {
+                    violation = false;
+                    console.log(`Account ${account} not in violation`);
+                } else {
+                    violation = true;
+                    console.log(`Account ${account} in violation`);
+                }
+
+                health_scores[account] = {
+                    health: ethers.utils.formatEther(health),
+                    violation: violation
+                };
+            }
+        }
+
+        let outputJson = JSON.stringify(health_scores);
+        fs.writeFileSync(`${filename}.json`, outputJson + "\n");
+    });
+
+task("gov:forkModuleInstall", "Impersonate installer admin on mainnet fork and install a module")
+    .addVariadicPositionalParam("addrs")
+    .setAction(async (addrs) => {
+        const {et, ctx, admin} = await setupGovernanceFork("installer");
+        await et.taskUtils.runTx(ctx.contracts.installer.connect(admin).installModules(addrs, await ctx.txOpts()));
+    });
+
+task("gov:forkSetPricingConfig", "Impersonate governor admin on mainnet fork and set pricing config for an asset")
```

Comment:
why not reuse the `gov:setAssetConfig`, `gov:setPricingConfig` and requested `gov: setChainlinkPriceFeed` instead of duplicating code in the `gov:fork...` tasks? it'd be enough to add an optional parameter to existing tasks that indicates whether the action should be executed on fork

Reply:
sounds good, I was considering this as well. I will do this @kasperpawlowski and implement the other feedback shortly.